### PR TITLE
add `use-t` rule (fixes #38)

### LINKED ISSUE #38: `use-t` rule
Rule proposal: Enforce that the test function has either no parameters, or only one parameter named `t`.

### Bad
```js
test(foo => { // Incorrect name
  // ...
});

test((foo) => { // Incorrect name
  // ...
});

test((t, bar) => { // too many arguments
  // ...
});

test((bar, t) => { // too many arguments
  // ...
});
```

### Good
```js
test(t => {
  // ...
});

test(foo); // As I don't think we can infer the arguments of foo, or even if it is a function

test.todo("foo");
```

I think this should be a recommended rule at error level, especially considering that all rules assume that `t` is indeed called `t`. What do you think?

### PR DESCRIPTION
**Pull request by jfmengels**
adds `use-t` rule (fixes #38) :)

### CODE REVIEW

File: docs/rules/use-t.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Ensure test functions use `t` as their parameter
+
+The convention is to have the parameter in AVA's test function be named `t`. Most rules in `eslint-plugin-ava` are based on the assumption.
```
Comment (sindresorhus) [inline docs/rules/use-t.md]: `the` => `that`

File: docs/rules/use-t.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+### Fail
+
+```js
+import test from 'ava';
+
+test(foo => { // Incorrect name
+	t.pass();
+});
+
+test((foo) => { // Incorrect name
```
Comment (sindresorhus) [inline docs/rules/use-t.md]: This is the same as the above, so I don't see the point in showing that.

**Comment by jfmengels** [comment]
:+1:. Updated

**Comment by sindresorhus** [comment]
Can you fix the merge conflict?

**Comment by jfmengels** [comment]
Done!

**Comment by sindresorhus** [comment]
tumblr_mpamnkvlcv1qdlh1io1_400

**Pull request merged**